dali/base/dadfs.cpp

```diff
@@ -5709,17 +5709,24 @@ class CDistributedSuperFile: public CDistributedFileBase<IDistributedSuperFile>
             target.setProp(prop, value);
     }
 
-    IDistributedFilePart &unprotectedQueryPart(unsigned idx)
+    IDistributedFilePart *unprotectedQueryPart(unsigned idx)
     {
-        if (subfiles.ordinality()==1)
-            return subfiles.item(0).queryPart(idx);
+        if (0 == subfiles.ordinality())
+            return nullptr;
+        else if ((1 == subfiles.ordinality()))
+        {
+            if (idx>=subfiles.item(0).numParts())
```

Comment:
Should have a comment about why this check is needed - initially I didn't understand it.